### ISSUE TITLE
Buttons

### ISSUE THREAD

**Issue by NJMannai** [dnet/androsphinx#18]
Device:  

- Nokia 7.1, Android 10, TalkBack 13. 
- Pixel 4a, Android 12, TalkBack 13. 

On the search screen, once you've entered a URL in the search box, there is a purple button with a white '+'. 
When navigating with Talkback, on the Nokia the button is detected as an actual button: 'Add a user, button', however on the Pixel phone it is not detected as a button, but an image: 'Add a user, image', which makes it impossible to double tap it and add a user.

It was difficult for us to figure out why it's a button on one phone and an image on the other phone.

Also, the ‘go back’ button next to the search box has no name. Users who are visually impaired don't know what the function of the button is. 

Related to WCAG 4.1.2 Name, role, value.

**Comment by dnet** [comment]
> however on the Pixel phone it is not detected as a button, but an image: 'Add a user, image',

That's weird, I tested it with a Pixel 5 using Android 12, now it runs Android 13 and it also says `Add a user, button`

That button is a `com.google.android.material.floatingactionbutton.FloatingActionButton` without any tinkering, so it should have the role correctly set by default.

> Also, the ‘go back’ button next to the search box has no name

On my Pixel 5 running Android 13, it says `Collapse, button`, the difference (and difficulty to do anything about it) is that the OS provides that search bar, not the app itself.

**Comment by NJMannai** [comment]
> On my Pixel 5 running Android 13, it says `Collapse, button`, the difference (and difficulty to do anything about it) is that the OS provides that search bar, not the app itself.

Ah ok I understand, so that would mean you would need to design a whole search box which is a lot of work.

In that case it is difficult for me to reject this button, however `collapse, button` is not correct since it’s not a collapsible button.

But you can leave it as it is.

> That's weird, I tested it with a Pixel 5 using Android 12, now it runs Android 13 and it also says `Add a user, button`
> That button is a com.google.android.material.floatingactionbutton.FloatingActionButton without any tinkering, so it should have the role correctly set by default.

This is very strange indeed. Me and my colleague however really did see this on our Pixel 4a (Android 12). I have to believe your word on it.

**Comment by dnet** [comment]
> `collapse, button` is not correct since it’s not a collapsible button

It's actually correct, as it collapses the search bar on the top. By default, you have

 - the title in the top left corner and
 - the magnifying glass icon in the top right corner.

If you click the magnifying glass icon,

 - the search bar expands, covering the title,
 - the magnifying glass moves to the left, tapping that reveals the search text entry field, and
 - a left arrow button appears on the left.
 
That left arrow button is not for back, but rather to collapse that search bar, returning the state to the first one, having the title on the left and the magnifying glass on the right. I think this fits the verb "collapse", the search bar collapses back into a single icon on the right.

**Comment by NJMannai** [comment]
That makes complete sense than, visually it looks like a back button.
Thank you for the feedback.
